extensions/asset/README.md

```diff
@@ -0,0 +1,34 @@
+# Asset Definition Specification
+
+**Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+
+A Collection extension to provide details about assets that are available in member Items.
```

Comment:
I think we should include more 'why' - what are the use cases for an implementor to make use of this. Not sure if this is the exact right place. But a bit more here would be good, and maybe flesh out more of it down below.